awx/api/serializers.py

```diff
@@ -2701,7 +2706,7 @@ def get_summary_fields(self, obj):
 class JobOptionsSerializer(LabelsListMixin, BaseSerializer):
 
     class Meta:
-        fields = ('*', 'job_type', 'inventory', 'project', 'playbook',
+        fields = ('*', 'job_type', 'inventory', 'project', 'playbook', 'scm_branch',
```

Comment:
This needs to be consistent... it's `scm_revision` elsewhere, which is more appropriate... it's more than just `branch`.

Reply:
`JobSerializer` has both fields `scm_branch` and `scm_revision`. The field name `scm_branch` was borrowed from the `Project` model.

The field `scm_revision` is the SHA1 hash or technical specifier of the commit that a job is ran against (like "r49" for subversion). This serves the purpose of audibility in AWX / Tower so that someone can look back in time and be sure they can recover the exact version of the playbook that was ran.

`scm_branch` is given by a user at some point. It can be several things, and the Ansible git module goes through fallback logic where it checks if it is (1) a branch (2) a tag (3) assume it's a commit, and possibly error. The fallback logic for evaluating an scm_branch here is modeled after their process.